src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -14,6 +14,9 @@ namespace System.Net.Http
     internal class WinInetProxyHelper
     {
         private bool _useProxy = false;
+        private bool _autoDetectionFailed;
+        private DateTime _lastTimeAutoDetectionFailed;
+        private TimeSpan _recentAutoDetectionInterval = new TimeSpan(0, 2, 0); // 2 minutes.
```

Comment:
I'm happy with 2 minutes if you think that's good.  I'd also be ok if it were more frequent, like 1 minute or even 30 seconds.  Just avoiding it on every request when making thousands of requests per second is great.